From 2d3a0d34be100ae4bf27d29e3d62a1843844b7c3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:33 -0400
Subject: [tools] perf/completion: Complete 'perf kvm'

Message-id: <1408953210-26343-146-git-send-email-jolsa@redhat.com>
Patchwork-id: 88862
O-Subject: [PATCH RHEL7.1 BZ1133083 145/322] perf completion: Complete 'perf kvm'
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 8f2f5ada719560954174da30ce0a67261c616e39
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Wed Dec 11 16:04:15 2013 +0530

description
===========
Currently, there is no way to enumerate the subcommands under 'perf
kvm', so hardcode them.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/perf-completion.sh b/tools/perf/perf-completion.sh
index 4949488..496e2ab 100644
--- a/tools/perf/perf-completion.sh
+++ b/tools/perf/perf-completion.sh
@@ -121,6 +121,10 @@ __perf_main ()
  elif [[ $prev == "-e" && "${words[1]}" == @(record|stat|top) ]]; then
   evts=$($cmd list --raw-dump)
   __perfcomp_colon "$evts" "$cur"
+ # List subcommands for 'perf kvm'
+ elif [[ $prev == "kvm" ]]; then
+  subcmds="top record report diff buildid-list stat"
+  __perfcomp_colon "$subcmds" "$cur"
  # List long option names
  elif [[ $cur == --* ]];  then
   subcmd=${words[1]}
-- 
1.7.1